### PR TITLE
Add ternary operator support to expression parser

### PR DESCRIPTION
**Pull request by mlschroe**

### CODE REVIEW

**Comment by pavlinamv** [comment]
I think that adding ternary operator support to the expression parser is useful and it will not cause problems in the parsing of the existing expressions. 

There is one thing that can be improved -cases like:
 ```
--eval '%{expr: 1 ?}'
--eval '%{expr: 0 ? 3 : }'
```

  returns an error, but without any message. It can be improved.

**Comment by mlschroe** [comment]
I modeled the code after the other operators. Things like `--eval '%{expr: 4 +}'` have the same problem.

**Comment by mlschroe** [comment]
I've opened another pull request that fixes the error printing as this is unrelated to the ternary operator.

**Comment by pavlinamv** [comment]
Thank you. It solves the problem.

**Comment by pavlinamv** [comment]
Could you change file
```
rpm/doc/manual/spec 
```

in section "Conditionals" the text after example:
```
%if 0%{?fedora} > 10 || 0%{?rhel} > 7
```

according of the PR?

**Comment by pmatilai** [comment]
@pavlinamv , what do you mean by changing those conditionals in the documentation? I don't see them needing changing because of this, but maybe I'm missing something.

**Comment by mlschroe** [comment]
I added the ternary operator to the list of supported operators.

**Comment by pavlinamv** [comment]
@mlschroe: Thank you. The PR looks good to me.

@pmatilai After the mentioned example there is a list of operators that can be used in a condition after
```
%if
```

File: rpmio/expression.c [hunk 1]
```diff
@@ -671,6 +682,52 @@ static Value doLogical(ParseState state)
...
+	goto err;
+    }
+    valueFree(v1);
+    if (rdToken(state))
+      goto err;
+    v1 = doTernary(state);
+    if (v1 == NULL)
+      goto err;
+    if (state->nextToken != TOK_TERNARY_ALT) {
+      exprErr(&state, _("syntax error in expression"), state->p);
```
Comment (pmatilai) [inline rpmio/expression.c]: expression.c: In function ‘doTernary’:
expression.c:712:15: warning: passing argument 1 of ‘exprErr’ from incompatible pointer type [-Wincompatible-pointer-types]
  712 |       exprErr(&state, _("syntax error in expression"), state->p);
      |               ^~~~~~

Comment (mlschroe) [reply rpmio/expression.c]: Damn. I'm copying from the wrong line ;)

**Comment by pmatilai** [comment]
Okay, looks nice and contained to me. Thanks a bunch!

**Pull request merged**